src/librustc_passes/check_const.rs

```diff
@@ -27,13 +27,15 @@ use std::fmt;
 enum NonConstExpr {
     Loop(hir::LoopSource),
     Match(hir::MatchSource),
+    OrPattern,
```

Comment:
cc @oli-obk @ecstatic-morse for awareness. This is a great catch I did not think of! 💖